packages/grafana-ui/src/components/Table/utils.ts

```diff
@@ -143,16 +148,26 @@ export function calculateUniqueFieldValues(rows: any[], field?: Field) {
   const set: Record<string, any> = {};
 
   for (let index = 0; index < rows.length; index++) {
-    const fieldIndex = parseInt(rows[index].id, 10);
-    const fieldValue = field.values.get(fieldIndex);
-    const displayValue = field.display ? field.display(fieldValue) : fieldValue;
-    const value = field.display ? formattedValueToString(displayValue) : displayValue;
-    set[value || '(Blanks)'] = fieldValue;
+    const value = rowToFieldValue(rows[index], field);
+    set[value || '(Blanks)'] = value;
   }
 
   return set;
 }
 
+export function rowToFieldValue(row: any, field?: Field): string {
+  if (!field || !row) {
+    return '';
+  }
+
+  const index = parseInt(row.id, 10);
```

Comment:
there is an index prop on a row here any reason not to use that?

Reply:
I didn't know that, is index a more correct way to access the __`column index`__?

Reply:
yes, react-table's Row has in index. But, to get the **column index** we usually keep a counter in the fields loop and pass that counter downstream.

Reply:
Thank you for clarifying, I've changed it to index instead.